dist/angular-promise-buttons.js

```diff
@@ -23,47 +23,110 @@ angular.module('angularPromiseButtons')
                 var cfg = providerCfg;
                 var promiseWatcher;
 
-
-                function handleLoading(btnEl)
+                function setLoadingState(btnEl)
                 {
                     if (cfg.btnLoadingClass && !cfg.addClassToCurrentBtnOnly) {
                         btnEl.addClass(cfg.btnLoadingClass);
                     }
                     if (cfg.disableBtn && !cfg.disableCurrentBtnOnly) {
                         btnEl.attr('disabled', 'disabled');
                     }
+                    if (cfg.btnLoadingHtml)
+                    {
+                        btnEl.html(cfg.btnLoadingHtml);
+                    }
+                }
+
+                function handleLoadingFinished(btnEl, defaultHtml, onEndFn, onEndConfig)
+                {
+                    removeLoadingState(btnEl);
+                    setFinishedState(btnEl, onEndConfig);
+
+                    //OnSuccess or OnError
+                    if (onEndFn && typeof onEndFn === 'function')
+                    {
+                        onEndFn();
+                    }
+
+                    if (cfg.onComplete && typeof cfg.onComplete === 'function')
+                    {
+                        cfg.onComplete();
+                    }
+
+                    var waitTime = 0;
+                    if (onEndConfig && onEndConfig.resultWaitTime && onEndConfig.resultWaitTime >= 0) {
+                        waitTime = onEndConfig.resultWaitTime;
+                    }
+
+                    return $timeout(function() {
```

Comment:
If you're using a $timeout you always should provide a cancelation if the the scope is destroyed, e.g.:
```
$scope.$on("$destroy", function( event ) {
  $timeout.cancel( timer );
});
```

see http://www.bennadel.com/blog/2548-don-t-forget-to-cancel-timeout-timers-in-your-destroy-events-in-angularjs.htm